libs/parser/parser.c

```diff
@@ -161,6 +166,21 @@ void parser_free(parser_t *p)
 }
 
 
+static int parser_headerNameCheck(const char *headerName)
+{
+	regmatch_t regmatch;
+
+	if (regexec(&parser_common.headerNameRegex, headerName, 1, &regmatch, 0) == 0) {
+		/* Check if whole string matches regex */
+		if (regmatch.rm_so == 0 && regmatch.rm_eo == strlen(headerName)) {
+			return 0;
+		}
+	}
+
+	return -1;
```

Comment:
Change pattern in line `445` to `"^" WORD "$"` and
```suggestion
	if (regexec(&parser_common.headerNameRegex, headerName, 1, &regmatch, 0) == 0) {
		return 0;
	}

	return -1;
```
Missing:
```
	regfree(&parser_common.headerNameRegex);
```
in `parser_regexFree(void)`